hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/MetadataCompactResolverTest.java

```diff
@@ -168,8 +172,21 @@ public void test_resolveMetadata(boolean key, String prefix) {
                 new MapTableField(prefix, QueryDataType.OBJECT, true, QueryPath.create(prefix))
         );
         assertThat(metadata.getQueryTargetDescriptor()).isEqualTo(GenericQueryTargetDescriptor.DEFAULT);
-        assertThat(metadata.getUpsertTargetDescriptor())
-                .isEqualToComparingFieldByField(new CompactUpsertTargetDescriptor("test"));
+        SchemaWriter schemaWriter = new SchemaWriter("test");
+        schemaWriter.addField(new FieldDescriptor("boolean", FieldType.BOOLEAN));
+        schemaWriter.addField(new FieldDescriptor("byte", FieldType.BYTE));
+        schemaWriter.addField(new FieldDescriptor("short", FieldType.SHORT));
+        schemaWriter.addField(new FieldDescriptor("int", FieldType.INT));
+        schemaWriter.addField(new FieldDescriptor("long", FieldType.LONG));
+        schemaWriter.addField(new FieldDescriptor("float", FieldType.FLOAT));
+        schemaWriter.addField(new FieldDescriptor("double", FieldType.DOUBLE));
+        schemaWriter.addField(new FieldDescriptor("decimal", FieldType.DECIMAL));
+        schemaWriter.addField(new FieldDescriptor("string", FieldType.UTF));
+        schemaWriter.addField(new FieldDescriptor("time", FieldType.TIME));
+        schemaWriter.addField(new FieldDescriptor("date", FieldType.DATE));
+        schemaWriter.addField(new FieldDescriptor("timestamp", FieldType.TIMESTAMP));
+        schemaWriter.addField(new FieldDescriptor("timestampTz", FieldType.TIMESTAMP_WITH_TIMEZONE));
+        assertEquals(metadata.getUpsertTargetDescriptor(), new CompactUpsertTargetDescriptor(schemaWriter.build()));
```

Comment:
Maybe it's worth adding tests for `resolveAndValidateFields` where we check all the fields are resolved correctly & that on `OBJECT` we fail.